### ISSUE TITLE
MemoryError when running CrossValTesting

### ISSUE THREAD

**Issue by queirozfcom** [scikit-multilearn/scikit-multilearn#39]
Hi Piotr. I was trying to run one of the tests you include in the project (CrossValTesting).

You can view it [on this link](https://gist.github.com/queirozfcom/abc0fd22dc77056d145e)

I even bought more RAM for my PC because  I thought I had too little but when I run this only about 3.5GB of m RAM is used, and I have 8GB total RAM so I don't think it's because of that.

Did this happen to you as well? Do you know what may be causing it? Thanks!

**Comment by ChristianSch** [comment]
Hi,

as this test is kinda from me I'll jump just in. I had the same problem, but only with the RCV1. I figured it was my fault due to having not enough memory for the data set (but I'm not sure). I also have 8GB. Did you try it with another dataset?

I'll see if I can run this with my bsc thesis dataset (also multi-label) which is much smaller and I already ran it on my MacBook with cross validation and this implementation.

**Comment by queirozfcom** [comment]
I will try with another dataset. Thing is, this dataset is (I think) in some kind of binary format, maybe python pickle or something like that. Probably arff files will work ok - Is `.arff` the default format for multilabel datasets? I keep seeing it.

**Comment by ChristianSch** [comment]
Never heard of it. I used csv or hdf5 depending on the dimensionality (try fitting tensors into csv rows :D). I had to interchange data between python and lua (torch) so I used hdf5. I'll look into .arff for the future, thanks.

**Comment by ChristianSch** [comment]
Hey, so I ran one of my thesis experiment setups again with the newest code (https://github.com/ChristianSch/thesis-code/blob/65294ed4a1303e4f7134d4acf0e26ddaa6116161/Baseline/Baseline%201%20-%20bow-tf-idf-doc2vec%20etc.ipynb, see `In[13]`) and it works. Sadly the dataset is NDA stuff, so I can't share it. I'll keep looking for multi-label datasets and shall run this again to see what's going on.
(Also, the code is really bad, sorry for that. I re-train them on evaluation due to the sklearn internals I didn't know at that time.)

**Comment by niedakh** [comment]
@queirozfcom it actually happened to me on a server with 131GB of RAM. For the data like rcv1 you should only use a sparse representation system. It is now in place, since 0.0.3 I think, but naive gaussian base classifiers dont handle sparse data for some reason, one needs to try the trees as base classifier.
